circuit/program/src/data/record/to_bits.rs

```diff
@@ -18,78 +18,70 @@ impl<A: Aleo> ToBits for Record<A, Plaintext<A>> {
+            identifier.write_bits_be(&mut data_bits_be);
+            entry.write_bits_be(&mut data_bits_be);
+        }
 
         // Construct the record bits.
-        let mut bits_be = self.owner.to_bits_be();
-        bits_be.extend(U32::constant(console::U32::new(data_bits_be.len() as u32)).to_bits_le());
-        bits_be.extend(data_bits_be);
-        bits_be.extend(self.nonce.to_bits_be());
-        bits_be
+        self.owner.write_bits_be(vec);
+        U32::constant(console::U32::new(data_bits_be.len() as u32)).write_bits_be(vec);
+        vec.extend_from_slice(&data_bits_be);
+        self.nonce.write_bits_be(vec);
     }
 }
 
 impl<A: Aleo> ToBits for Record<A, Ciphertext<A>> {
     type Boolean = Boolean<A>;
 
     /// Returns this data as a list of **little-endian** bits.
-    fn to_bits_le(&self) -> Vec<Self::Boolean> {
+    fn write_bits_le(&self, vec: &mut Vec<Self::Boolean>) {
         // Compute the data bits.
-        let data_bits_le = self
-            .data
-            .iter()
-            .flat_map(|(identifier, entry)| [identifier.to_bits_le(), entry.to_bits_le()])
-            .flatten()
-            .collect::<Vec<_>>();
+        let mut data_bits_le = vec![];
+        for (identifier, entry) in &self.data {
+            identifier.write_bits_le(&mut data_bits_le);
+            entry.write_bits_le(&mut data_bits_le);
+        }
 
         // Construct the record bits.
-        let mut bits_le = self.owner.to_bits_le();
-        bits_le.extend(U32::constant(console::U32::new(data_bits_le.len() as u32)).to_bits_le());
-        bits_le.extend(data_bits_le);
-        bits_le.extend(self.nonce.to_bits_le());
-        bits_le
+        self.owner.write_bits_le(vec);
+        U32::constant(console::U32::new(data_bits_le.len() as u32)).write_bits_le(vec);
+        vec.extend_from_slice(&data_bits_le);
+        self.nonce.write_bits_le(vec);
     }
 
     /// Returns this data as a list of **big-endian** bits.
-    fn to_bits_be(&self) -> Vec<Self::Boolean> {
+    fn write_bits_be(&self, vec: &mut Vec<Self::Boolean>) {
         // Compute the data bits.
-        let data_bits_be = self
-            .data
-            .iter()
-            .flat_map(|(identifier, entry)| [identifier.to_bits_be(), entry.to_bits_be()])
-            .flatten()
-            .collect::<Vec<_>>();
+        let mut data_bits_be = vec![];
+        for (identifier, entry) in &self.data {
+            identifier.write_bits_be(&mut data_bits_be);
+            entry.write_bits_be(&mut data_bits_be);
+        }
 
         // Construct the record bits.
-        let mut bits_be = self.owner.to_bits_be();
-        bits_be.extend(U32::constant(console::U32::new(data_bits_be.len() as u32)).to_bits_le());
-        bits_be.extend(data_bits_be);
-        bits_be.extend(self.nonce.to_bits_be());
-        bits_be
+        self.owner.write_bits_be(vec);
+        U32::constant(console::U32::new(data_bits_be.len() as u32)).write_bits_be(vec);
```

Comment:
same note as [here](https://github.com/AleoHQ/snarkVM/pull/1836#discussion_r1272477351) - this is a change, as it seems like a copy-paste error in the current code